src/components/PopupWithForm.js

```diff
@@ -26,4 +29,12 @@ export default class PopupWithForm extends Popup {
     });
     super.setEventListeners();
   }
+
```

Comment:
Needs correction: (line 28) Сlose modal in the case of successful api response in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.

Reply:
This has been corrected.